include/deal.II/base/parsed_convergence_table.h

```diff
@@ -0,0 +1,770 @@
+ * via a parameter file.
+ *
+ * With a small modification, the same code can be used to estimate the errors
+ * of mixed or multi-physics problems, e.g.:
+ * @code
+ * using namespace ParsedConvergenceTableFlags;
+ * ParsedConvergenceTable table({"u,u,p"},{{H1, L2}, {L2}});
+ *
+ * ParameterHandler prm;
+ * table.add_parameters(prm);
+ *
+ * for (unsigned int i = 0; i < n_cycles; ++i)
+ *   {
+ *     ... // do some computations
+ *     table.error_from_exact(dof_handler, solution, exact_solution);
+ *   }
+ * table.output_table(std::cout);
+ * @endcode
+ *
+ * The above code assumes that you are solving a Stokes problem with three
+ * components. Two components for the vector velocity field `u`, and one
+ * component for the pressure field `p`, and will produce an error
+ * table with `H1` and `L2` norm of the error in the velocity field (first two,
+ * components) and `L2` error in the pressure field.
+ *
+ * By calling the method add_parameters() passing a ParameterHandler object,
+ * the following options will be defined in the given ParameterHandler object,
+ * and can be modified at run time through a parameter file:
+ * @code
+ * set Enable computation of the errors = true
+ * set Enable output to streams         = true
+ * set Error file name                  =
+ * set Error precision                  = 3
+ * set Extra columns                    = dofs, cells
+ * set List of error norms to compute   = Linfty, L2, H1
+ * set Rate key                         = dofs
+ * @endcode
+ *
+ * When using this class, please cite
+ * @code{.bib}
+ * @article{SartoriGiulianiBardelloni-2018-a,
+ * 	Author = {Sartori, Alberto and Giuliani, Nicola and
+ *            Bardelloni, Mauro and Heltai, Luca},
+ * 	Journal = {SoftwareX},
+ * 	Pages = {318--327},
+ * 	Title = {{deal2lkit: A toolkit library for high performance
+ *            programming in deal.II}},
+ *  Doi = {10.1016/j.softx.2018.09.004},
+ * 	Volume = {7},
+ * 	Year = {2018}}
+ * @endcode
+ *
+ * @author Luca Heltai, 2019.
+ */
+class ParsedConvergenceTable
+{
+public:
+  /**
+   * Minimal constructor for ParsedConvergenceTable objects.
+   *
+   * The number of components must match the number of components of the
+   * finite element space that is used to compute the errors. If
+   * a component name is repeated, than it is interpreted as a vector field,
+   * and the errors of the repeated components are grouped together.
+   *
+   * The size of the vector @p list_of_error_norms must match the number of
+   * unique component names, and may contain zero or more comma separated
+   * strings that identify the norm types to compute for each component.
+   *
+   * For example, the following constructor
+   * @code
+   * using namespace ParsedConvergenceTableFlags;
+   * ParsedConvergenceTable table({"u", "v", "v"}, {{Linfty}, {L2, H1}});
+   * @endcode
+   *
+   * Will produce (if the parameter file is left untouched) the following table:
```

Comment:
remove the empty line and apply this to make it a complete sentence:
```suggestion
   * will produce (if the parameter file is left untouched) the following table:
```